data_model/derive/tests/event_set.rs

```diff
@@ -0,0 +1,139 @@
+mod events {
+    use iroha_data_model_derive::EventSet;
+    #[derive(EventSet)]
+    pub enum TestEvent {
+        Event1,
+        Event2,
+        NestedEvent(AnotherEvent),
+    }
+
+    pub struct AnotherEvent;
+}
+
+use events::{AnotherEvent, TestEvent, TestEventSet};
+use serde_json::json;
+
+#[test]
+fn serialize() {
+    assert_eq!(
+        serde_json::to_value(&TestEventSet::Event1).unwrap(),
+        json!(["Event1"])
+    );
+    assert_eq!(
+        serde_json::to_value(&(TestEventSet::Event1 | TestEventSet::Event2)).unwrap(),
+        json!(["Event1", "Event2"])
+    );
+    assert_eq!(
+        serde_json::to_value(&(TestEventSet::Event1 | TestEventSet::AnyNestedEvent)).unwrap(),
+        json!(["Event1", "AnyNestedEvent"])
+    );
+    assert_eq!(
+        serde_json::to_value(&TestEventSet::all()).unwrap(),
+        json!(["Event1", "Event2", "AnyNestedEvent"])
+    );
+}
+
+#[test]
+fn deserialize() {
```

Comment:
We cover with tests cases like:
1. empty: []
2. duplication ["Event1", "Event1"]
3. wrong variants are rejected

Reply:
Fair, will add those

Reply:
done